### PR TITLE
feat: add azure provider templates 

### PR DESCRIPTION
**Pull request by livio-a**
Adds possibility to manage and use Microsoft Azure template based providers

```[tasklist]
### Definition of Ready
- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

File: internal/query/projection/idp_template_test.go [hunk 1]
```diff
@@ -410,6 +410,270 @@ func TestIDPTemplateProjection_reducesOAuth(t *testing.T) {
...
+		event func(t *testing.T) eventstore.Event
+	}
+	tests := []struct {
+		name   string
+		args   args
+		reduce func(event eventstore.Event) (*handler.Statement, error)
+		want   wantReduce
+	}{
+		{
+			name: "instance reduceAzureADIDPAdded",
```
Comment (stebenz) [inline internal/query/projection/idp_template_test.go:424]: would add a minimal test as some attributes can be null in the database, at least for either org or instance

File: proto/zitadel/admin.proto [hunk 2]
```diff
@@ -1320,6 +1320,30 @@ service AdminService {
...
+
+        option (zitadel.v1.auth_option) = {
+            permission: "iam.idp.write"
+        };
+    }
+
+    // Change an existing Azure AD identity provider on the instance
+    rpc UpdateAzureADProvider(UpdateAzureADProviderRequest) returns (UpdateAzureADProviderResponse) {
+        option (google.api.http) = {
+            post: "/idps/azure/{id}"
```
Comment (stebenz) [inline proto/zitadel/admin.proto:1338]: Shouldn't that be a put?

File: proto/zitadel/management.proto [hunk 3]
```diff
@@ -6536,6 +6536,30 @@ service ManagementService {
...
+
+        option (zitadel.v1.auth_option) = {
+            permission: "org.idp.write"
+        };
+    }
+
+    // Change an existing Azure AD identity provider in the organisation
+    rpc UpdateAzureADProvider(UpdateAzureADProviderRequest) returns (UpdateAzureADProviderResponse) {
+        option (google.api.http) = {
+            post: "/idps/azure/{id}"
```
Comment (stebenz) [inline proto/zitadel/management.proto:6554]: Shouldn't that be a put?

**Comment by peintnermax** [comment]
Seems to me that the tenantId is not correctly set, neither on add nor update

**Review by stebenz — DISMISSED** [review]

**Review by peintnermax — APPROVED** [review]
